src/JarvisDesk.cpp

```diff
@@ -107,26 +104,46 @@ class JarvisDesk_impl {
     }
     else pending_stop = false;
 
-    if (latch_timer.trigger()) {
+    if (latch_timer.trigger())
       unlatch();
-    }
 
     // Publish pending updates periodically to AdafruitIO
     io_send();
   }
 
+  void press_Memory(int duration = 30) {
```

Comment:
I figured out that HS0+HS3 isn't the memory button after all.  It's the "wake sequence" to get the controller to start sending height info.  The "Memory" button doesn't actually send anything on my Jarvis desk.  The handset waits for the "preset" button to be pressed, and then it sends the PROGRAM_PRESET command via serial. But the HS0+HS3 is also useful to interrupt the desk from moving, which is all it's needed for here.   Does yours also act this way?

Reply:
memory will stop the desk from working, but it only uses one line. This handset basically never sends anything over UART. When moving for longer periods of times, it will have an almost random sequence that is sent from the handset to the controller. The controller then echos the same values back (even if it is mid "broadcast").

Reply:
The handset sends an almost random sequence on the UART?  Have you checked to see if it's actually serial data?  Maybe those lines are being used for something else with your controller.  Do you have a logic analyzer that can capture the line signals between the HS and Controller?

Reply:
I do have a cheap logic analyzer I can set up. The only time the Handset sends anything is when the desk is in motion for a longer amount of time (going to a preset or holding up/down). The controller sends back exactly what the handset sent no matter what it is in the middle of sending at the moment. So, I could get a sequence of 1 1 FF 0 FC ( height with a 255 handset echo back in the middle). This language does not appear to be regular so an FA/State Machine is difficult (or maybe even impossible) to model the protocol completely.

I will do some captures this weekend and will try and get as much as I can captured and labeled.